tests/test_runrms.py

```diff
@@ -58,6 +59,167 @@ def test_scan_rms(tmpdir):
+    assert runner.version_fromproject is None
+
+    # Add version to the .master:
+    dot_master.write_text("release : 18.0.0")
+    runner.scan_rms()
+    assert runner.version_fromproject == "18.0.0"
+
+    # Add version in wrong place to the .master:
+    dot_master.write_text("End GEOMATIC\nrelease : 18.0.0")
+    runner = rr.RunRMS()
+    runner.project = mocked_rms
+    runner.scan_rms()
+    assert runner.version_fromproject is None
+
+    # Test mkeys:
+    dot_master.write_text(
+        "fileversion : foo\nuser : foobert\nvariant : bogus bogus bogus"
+    )
+    runner = rr.RunRMS()
+    runner.project = mocked_rms
+    runner.scan_rms()
+    assert runner.fileversion == "foo"
+    assert runner.user == "foobert"
+    assert runner.variant == "unknown"  # Too many strings provided in .master
+
+
+def test_runlogger(tmpdir):
+    tmpdir.chdir()
+
+    runner = rr.RunRMS()
+    runner.runloggerfile = "not-existing"
+    # Nothing happens, skipped because it does not exist
+    runner.runlogger()
+
+    # Make an empty file and prove that it will be logged to:
+    Path("foo-log").write_text("")
+    assert not Path("foo-log").read_text()
+    runner.runloggerfile = "foo-log"
+    runner.runlogger()
+    assert Path("foo-log").read_text()
+
+
+@pytest.mark.parametrize(
+    "os_id, expected",
+    [
+        ("Red Hat Enterprise Linux Server release 6.0 (bar)", "x86_64_RH_6"),
+        ("Red Hat Enterprise Linux Server release 7.2 (Maipo)", "x86_64_RH_7"),
+        ("Red Hat Enterprise Linux Server release 8.2 (foo)", "x86_64_RH_8"),
+        pytest.param("foobar", None, marks=pytest.mark.xfail(raises=IndexError)),
+    ],
+)
+def test_detect_os(os_id, expected, tmpdir, mocker):
+    tmpdir.chdir()
+    release_file = Path("redhat-release")
+    release_file.write_text(os_id)
+    mocker.patch("subscript.runrms.runrms.RHEL_ID", release_file)
+    runner = rr.RunRMS()
+    assert runner.osver == expected
+
+
+def test_detect_os_default(tmpdir, mocker):
+    tmpdir.chdir()
+    release_file = Path("not-existing-file")
+    mocker.patch("subscript.runrms.runrms.RHEL_ID", release_file)
+    runner = rr.RunRMS()
+    assert runner.osver == "x86_64_RH_7"
+
+
+def test_store_pythonpath(mocker):
+    mocker.patch("os.environ", {"PYTHONPATH": "foo/bar/com"})
+    runner = rr.RunRMS()
+    assert "foo/bar/com" in runner.oldpythonpath
+    assert runner.pythonpath is None
+
+    mocker.patch("os.environ", {"PYTHONbarfPATH": "foo/bar/com"})
+    runner = rr.RunRMS()
+    assert runner.oldpythonpath == ""
+    assert runner.pythonpath is None
+
+
+def test_store_rmspluginpath(mocker):
+    mocker.patch("os.environ", {"RMS_PLUGINS_LIBRARY": "foo/bar/com"})
+    runner = rr.RunRMS()
+    assert "foo/bar/com" in runner.oldpluginspath
+    assert runner.pluginspath is None
+
+    mocker.patch("os.environ", {"PYTHONbarfPATH": "foo/bar/com"})
+    runner = rr.RunRMS()
+    assert runner.oldpythonpath == ""
+    assert runner.pythonpath is None
```

Comment:
Is `PYTHONbarfPATH` intentional or typo?

Reply:
It is intentional, as a dummy environment where nothing ends up in oldpythonpath.

Reply:
OK, it looked like a paste error; perhaps `PYTHONdummyPATH` would be more obvious?

Reply:
fixed!